### ISSUE TITLE
Source Control window does not refresh after staging with right-click menu in worktrees

### ISSUE THREAD

**Issue by tsangwpx** [Microsoft/vscode#68507]
- VSCode Version: 1.32.0-insider (user setup)
- OS Version: Windows_NT x64 10.0.17763
- Git for Windows: git version 2.20.1.windows.1

Source control does not work in a git worktree as it does in the main worktree 

Steps to Reproduce:
1. In an empty directory, run the setup script to create the `master` worktree and the `fork` worktree
```
a2345=$'a\n2\n3\n4\n5'
a123z=$'a\n1\n2\n3\nz'

git init master
cd master
echo "$a2345" > helloworld.txt
git add .
git commit -m "initial commit"

git worktree add -b fork ../fork
echo "$a123z" > helloworld.txt
echo "../fork/.git" > .git/worktrees/fork/gitdir
cd ../fork
echo "gitdir: ../master/.git/worktrees/fork" > .git
echo "$a123z" > helloworld.txt
```

Note 1: Using relative path in gitdir make both WSL and Git on Windows works.
Note 2: I tired using absolute path in gitdir, it does not work neither.

2.
Expected:
Open VS Code in the `fork` directory, and stage the changed lines one by one using the context menu. The diff window refresh itself. Everything is staged then.

Actual:
Open VS Code in the `fork` directory, and stage the changed lines one by one using the context menu. The diff window does not refresh itself. Only the last staging command was performed.

Does this issue occur when all extensions are disabled?: Yes

See the recording for the differences between the `master` worktree and the `fork` worktree`

[recording.zip](https://github.com/Microsoft/vscode/files/2855028/recording.zip)

**Comment by tsangwpx** [comment]
TLDR: vscode do refresh itself when files are changed in the main worktree but do not in the `git worktree add` worktree.

**Comment by joaomoreno** [comment]
Caused by #68038
